nmigen/lib/cdc.py

```diff
@@ -1,7 +1,15 @@
 from .. import *
+from math import gcd
 
 
-__all__ = ["MultiReg", "ResetSynchronizer"]
+__all__ = ["MultiReg", "ResetSynchronizer", "PulseSynchronizer", "Gearbox"]
+
+
+def _incr(signal, modulo):
+    if modulo == 2 ** len(signal):
+        return signal + 1
+    else:
+        return Mux(signal == modulo - 1, 0, signal + 1)
```

Comment:
There's already one in `lib.fifo` and this is getting ridiculous.

Reply:
Agreed, it should be either inlined at my callsites, or preferably factored out. Wanted to keep the scope of this PR well-defined though.

Reply:
Can you think of a nicer place to put this? It's a common pattern (as is saturating increment/decrement)